apps/bfd-pipeline/bfd-pipeline-ccw-rif/src/main/java/gov/cms/bfd/pipeline/ccw/rif/load/RifLoader.java

```diff
@@ -354,24 +375,23 @@ public Integer getValue() {
    * @param postgresBatch the {@link PostgreSqlCopyInserter} for the current set of {@link
    *     RifFilesEvent}s being processed
    * @param resultHandler the {@link Consumer} to notify when the batch completes successfully
-   * @param errorHandler the {@link Consumer} to notify when the batch fails for any reason
    */
   private void processAsync(
       BlockingThreadPoolExecutor loadExecutor,
       List<RifRecordEvent<?>> recordsBatch,
       long loadedFileId,
       PostgreSqlCopyInserter postgresBatch,
-      Consumer<RifRecordLoadResult> resultHandler,
-      Consumer<Throwable> errorHandler) {
+      Consumer<RifRecordLoadResult> resultHandler) {
     loadExecutor.submit(
         () -> {
           try {
             List<RifRecordLoadResult> processResults =
                 process(recordsBatch, loadedFileId, postgresBatch);
             processResults.forEach(resultHandler::accept);
           } catch (Throwable e) {
-            LOGGER.error("Error caught when processing async batch!");
-            errorHandler.accept(e);
+            LOGGER.error("Error caught when processing async batch!", e);
+            // Cannot use errorHandler here, as it will cause a long wait before shutdown
+            fatalFailure = true;
```

Comment:
Using the errorHandler here was causing a problem. The callback in this subthread caused a shutdown which blocked the subthread from returning until the shutdown was complete. The shutdown was waiting for all jobs to finish, and RifLoader was one of the jobs, which was waiting on its subthreads, and this meant we were deadlocked. The job would sit waiting on the 72 hour timeout on line 330, after which the job would finally give up waiting and we could finish application shutdown, however I doubt anyone actually waited 2 days to watch it gracefully shut down.